### PR TITLE
Merging redesign into master

### PR DESCRIPTION
**Pull request by aravindr93**
- mujoco version upgrade to 2.0
- gym version upgrade to 0.13 (pinned to this version)
- setup.py requirements updated to mujoco-py >= 2.0
- bonus rewards are added by default to the environments
- get_state and set_state functions updated to be consistent with trajopt

### CODE REVIEW

File: mj_envs/hand_manipulation_suite/assets/DAPG_pen.xml [hunk 1]
```diff
@@ -65,7 +65,7 @@
             <joint name="OBJRx" pos="0 0 0" axis="1 0 0" limited="false" damping="0" />
             <joint name="OBJRy" pos="0 0 0" axis="0 1 0" limited="false" damping="0" />
             <joint name="OBJRz" pos="0 0 0" axis="0 0 1" limited="false" damping="0" />
-            <geom name="pen" type="cylinder" size="0.015 0.065" condim="4" rgba=".6 .6 .6 .6" />
+            <geom name="pen" type="cylinder" size="0.015 0.065" condim="4" rgba=".6 .6 .6 .6" density="1500" />
```
Comment (vikashplus) [inline mj_envs/hand_manipulation_suite/assets/DAPG_pen.xml]: Changing mass can change the final policy's performance. Have we checked if this is a no-op wrt performance?

File: mj_envs/hand_manipulation_suite/pen_v0.py [hunk 2]
```diff
@@ -118,15 +127,15 @@ def ss(self, state_dict):
...
         self.sim.forward()
-        self.viewer.cam.distance = 1.2
+        self.viewer.cam.distance = 1.0
 
     def evaluate_success(self, paths):
         num_success = 0
         num_paths = len(paths)
         for path in paths:
-            if np.sum(path['rewards']) > 3000:
+            if np.sum(path['env_infos']['goal_achieved']) > 20:
```
Comment (vikashplus) [inline mj_envs/hand_manipulation_suite/pen_v0.py]: add a comment explaining what you are considering as success.

File: mj_envs/hand_manipulation_suite/relocate_v0.py [hunk 3]
```diff
@@ -13,10 +15,10 @@ def __init__(self):
         mujoco_env.MujocoEnv.__init__(self, curr_dir+'/assets/DAPG_relocate.xml', 5)
         
         # change actuator sensitivity
-        self.sim.model.actuator_gainprm[self.sim.model.actuator_name2id('A_WRJ1'):self.sim.model.actuator_name2id('A_WRJ0')+1,:] = np.array([10, 0, 0])
-        self.sim.model.actuator_gainprm[self.sim.model.actuator_name2id('A_FFJ3'):self.sim.model.actuator_name2id('A_THJ0')+1,:] = np.array([1, 0, 0])
-        self.sim.model.actuator_biasprm[self.sim.model.actuator_name2id('A_WRJ1'):self.sim.model.actuator_name2id('A_WRJ0')+1,:] = np.array([0, -10, 0])
-        self.sim.model.actuator_biasprm[self.sim.model.actuator_name2id('A_FFJ3'):self.sim.model.actuator_name2id('A_THJ0')+1,:] = np.array([0, -1, 0])
+        self.sim.model.actuator_gainprm[self.sim.model.actuator_name2id('A_WRJ1'):self.sim.model.actuator_name2id('A_WRJ0')+1,:3] = np.array([10, 0, 0])
+        self.sim.model.actuator_gainprm[self.sim.model.actuator_name2id('A_FFJ3'):self.sim.model.actuator_name2id('A_THJ0')+1,:3] = np.array([1, 0, 0])
```
Comment (vikashplus) [inline mj_envs/hand_manipulation_suite/relocate_v0.py]: It's a little bizarre that we need to relax the actuator only for FF knuckle. Either this indicates that we should do it for all knuckles or its a typo. I'm inclined to believe that this is the later -- (a) policies are less likely to be sensitive to knuckles actuator strength (b) Arm (ARJ3) is more likely to have increased sensitivity.

File: mj_envs/hand_manipulation_suite/assets/DAPG_pen.xml (hunk 1 above)
Comment (aravindr93) [reply mj_envs/hand_manipulation_suite/assets/DAPG_pen.xml]: Yes, I have tested it. The committed policies in hand_dapg are for this mass setting. Heavier pen gives smoother and better movements.

File: mj_envs/hand_manipulation_suite/relocate_v0.py (hunk 3 above)
Comment (aravindr93) [reply mj_envs/hand_manipulation_suite/relocate_v0.py]: Is this related to the original version or my updated version? The only change from the original is that in mujoco 1.5, there are 3 columns in the actuator fields, whereas in mujoco 2.0 there are 10 fields. Thus the columns are modified from `:` to `:3`. The names of actuators and values themselves are unchanged in this PR.

File: mj_envs/hand_manipulation_suite/pen_v0.py (hunk 2 above)
Comment (aravindr93) [reply mj_envs/hand_manipulation_suite/pen_v0.py]: Added a commit to address this.

File: mj_envs/hand_manipulation_suite/relocate_v0.py (hunk 3 above)
Comment (vikashplus) [reply mj_envs/hand_manipulation_suite/relocate_v0.py]: This is related to the original version, to begin with. My hypothesis is that either we don't need to update A_FFJ3 at all, or its a typo for A_ARJ3

Comment (aravindr93) [reply mj_envs/hand_manipulation_suite/relocate_v0.py]: I see, can you try both versions and let me know? The train script in hand_dapg should be super handy for this. You can just run it with the two versions, and we can finalize after that. In any case, the version in the PR gives good results and movements.

**Pull request merged**